common/hyperscaler/rules/rule.go

```diff
@@ -0,0 +1,358 @@
+package rules
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+
+	"github.com/kyma-project/kyma-environment-broker/internal/broker"
+)
+
+type Attribute struct {
+	Name          string
+	Description   string
+	setter        func(*Rule, string) (*Rule, error)
+	getter        func(*Rule) string
+	input         bool
+	output        bool
+	modifiedLabel string
+
+	modifiedLabelName string
+	ApplyLabel        func(r *Rule, labels map[string]string) map[string]string
+}
+
+func (a Attribute) String(r *Rule) any {
+	val := a.getter(r)
+	output := ""
+	if val == "true" {
+		output += fmt.Sprintf("%s, ", a.Name)
+	} else if val != "" && val != "false" {
+		output += fmt.Sprintf("%s=%s, ", a.Name, val)
+	}
+
+	return output
+}
+
+const (
+	PR_ATTR_NAME = "PR"
+	HR_ATTR_NAME = "HR"
+	EU_ATTR_NAME = "EU"
+	S_ATTR_NAME  = "S"
+
+	HYPERSCALER_LABEL = "hyperscalerType"
+	EUACCESS_LABEL    = "euAccess"
+	SHARED_LABEL      = "shared"
+
+	ASTERISK = "*"
+)
+
+var InputAttributes = []Attribute{
+	{
+		Name:        PR_ATTR_NAME,
+		Description: "Platform Region",
+		setter:      setPlatformRegion,
+		getter:      func(r *Rule) string { return r.PlatformRegion },
+		input:       true,
```

Comment:
Do we need these now?
Do we use it, is not it against YAGNI principle?

Reply:
I have found the usage.
We compute  max allowed number of input attributes.
Still we need to discuss it directly.